tests/model/test_model.py

```diff
@@ -1259,6 +1260,87 @@ def test_can_user_edit_topic(
         else:
             report_error.assert_called_once_with(expected_response[user_type][0])
 
+    @pytest.mark.parametrize(
+        "topic_name, msg_response, server_feature_level, topic_editing_limit_seconds,"
+        " expected_new_topic_name, expected_footer_error",
+        [
+            case(
+                "hi!",
+                {
+                    "subject": "hi!",
+                    "timestamp": 11662271397,
+                    "id": 1,
+                },
```

Comment:
This dict appears identical in each case, except for the time, so let's build it in the function to keep it simpler? You could test for the `None` return value case, but that may be simpler in a separate test.